src/foam/nanos/boot/Boot.java

```diff
@@ -58,25 +61,21 @@ public Boot(String datadir) {
 
     for ( int i = 0 ; i < l.size() ; i++ ) {
       NSpec sp = (NSpec) l.get(i);
+      XFactory factory = new SingletonFactory(new NSpecFactory((ProxyX) root_, sp));
+      factories_.put(sp.getName(), factory);
       logger.info("Registering:", sp.getName());
-      root_.putFactory(sp.getName(), new SingletonFactory(new NSpecFactory((ProxyX) root_, sp)));
+      root_.putFactory(sp.getName(), factory);
     }
 
+    X rootX = root_;
     serviceDAO_.listen(new AbstractSink() {
       @Override
       public void put(Object obj, Detachable sub) {
         NSpec sp = (NSpec) obj;
-        FObject newService = sp.getService();
-
-        if ( newService != null ) {
-          logger.info("Updating service configuration: ", sp.getName());
 
-          FObject service = (FObject) root_.get(sp.getName());
-          List<PropertyInfo> props = service.getClassInfo().getAxioms();
-          for (PropertyInfo prop : props) {
-            prop.set(service, prop.get(newService));
-          }
-        }
+        logger.info("Reload service factory:", sp.getName());
+        ((SingletonFactory) factories_.get(sp.getName()))
```

Comment:
Cast to an NSpecFactory and call invalidate() instead.

Reply:
Outdated.